src/client/src/producer.rs

```diff
@@ -25,6 +25,45 @@ impl TopicProducer {
         Self { topic, pool }
     }
 
+    /// Sends a key/value record to this producer's Topic.
+    ///
+    /// The partition that the record will be sent to is derived from the Key.
+    ///
+    /// # Example
+    ///
+    /// ```
+    /// # use fluvio::{TopicProducer, FluvioError};
+    /// # async fn example(producer: &TopicProducer) -> Result<(), FluvioError> {
+    /// producer.send("Key", "Value").await?;
+    /// # Ok(())
+    /// # }
+    /// ```
+    #[instrument(
+        skip(self, key, value),
+        fields(topic = %self.topic),
+    )]
+    pub async fn send<K, V>(&self, key: K, value: V) -> Result<(), FluvioError>
+    where
+        K: AsRef<[u8]>,
+        V: AsRef<[u8]>,
+    {
+        let key = key.as_ref();
+        let value = value.as_ref();
+        debug!(
+            key_size = key.len(),
+            value_size = value.len(),
+            "sending records:"
+        );
+
+        let replica = ReplicaKey::new(&self.topic, 0);
+        let spu_client = self.pool.create_serial_socket(&replica).await?;
```

Comment:
This is not optimized.  But looks this was previous implementation.  let's create issue to optimize this (not creating unnecessary virtual connection every time)

Reply:
https://github.com/infinyon/fluvio/issues/840